### PR TITLE
File pointers

### PR DESCRIPTION
**Pull request by stripes416**
Convert all usages of C FILE* to C++ std::ifstream / std::ofstream in libyami-utils/tests/

Libyami #744

### CODE REVIEW

File: tests/v4l2encode.cpp [hunk 1]
```diff
@@ -147,23 +148,23 @@ bool feedOneInputFrame(int fd, int index = -1 /* if index is not -1, simple enqu
 
 bool writeOneOutputFrame(uint8_t* data, uint32_t dataSize)
 {
-    static FILE* fp = NULL;
+    std::ifstream m_ofs = NULL;
```
Comment (uartie) [inline tests/v4l2encode.cpp]: don't assign NULL to non-pointer type

File: tests/v4l2encode.cpp [hunk 2]
```diff
@@ -147,23 +148,23 @@ bool feedOneInputFrame(int fd, int index = -1 /* if index is not -1, simple enqu
...
 
-    if(!fp) {
-        fp = fopen(outputFileName, "w+");
-        if (!fp) {
+    if(!m_ofs) {
```
Comment (uartie) [inline tests/v4l2encode.cpp]: Need to check `if(!m_ofs.is_open())` here.  `!m_ofs` only checks if failbit or badbit is set... afaik, these bits aren't set on a default constructed ofstream.  The only time you should use `!m_ofs` is after attempting to open a file or an operation has been performed on an opened file.

**Review by uartie — CHANGES_REQUESTED** [review]
I've rebased my branch with upstream.  Please rebase you branch with mine along with addressing my comments.

File: tests/v4l2encode.cpp (hunk 2 above)
Comment (stripes416) [reply tests/v4l2encode.cpp]: gtk!

File: tests/v4l2encode.cpp [hunk 3]
```diff
@@ -147,23 +148,23 @@ bool feedOneInputFrame(int fd, int index = -1 /* if index is not -1, simple enqu
 
 bool writeOneOutputFrame(uint8_t* data, uint32_t dataSize)
 {
-    static FILE* fp = NULL;
+    static std::ifstream m_ofs;
```
Comment (stripes416) [inline tests/v4l2encode.cpp]: I noticed just now that the existing code had the file marked as 'static' and I hadn't added that. Not a good combo when you open with 'trunc' enabled.

Comment (uartie) [reply tests/v4l2encode.cpp]: This should be a `std::ofstream`... and static just means it'll stay open until the program ends or until it's explicitly closed.

**Review by uartie — CHANGES_REQUESTED** [review]

File: tests/v4l2encode.cpp (hunk 3 above)
Comment (uartie) [reply tests/v4l2encode.cpp]: hmmm.. yes I see now that it is closed explicitly on isOutputEOS.  I'm not sure what the original author's intention is here... but your refactor looks logically equivalent (with std::ofstream change of course)

Comment (stripes416) [reply tests/v4l2encode.cpp]: Yeah, I don't understand why some of the things in this file were done the way they were, but I can roll with it. :)

**Pull request merged**